src/mito2/src/sst/index.rs

```diff
@@ -12,159 +12,155 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-pub(crate) mod applier;
-mod codec;
-pub(crate) mod creator;
+mod indexer;
 pub(crate) mod intermediate;
+pub(crate) mod inverted_index;
 pub(crate) mod puffin_manager;
+mod statistics;
 mod store;
 
 use std::num::NonZeroUsize;
 
 use common_telemetry::{debug, warn};
-use creator::SstIndexCreator;
-use object_store::ObjectStore;
+use puffin::puffin_manager::PuffinManager;
+use puffin_manager::{SstPuffinManager, SstPuffinWriter};
+use statistics::{ByteCount, RowCount};
 use store_api::metadata::RegionMetadataRef;
-use store_api::storage::RegionId;
+use store_api::storage::{ColumnId, RegionId};
 
+use crate::access_layer::OperationType;
+use crate::config::InvertedIndexConfig;
 use crate::metrics::INDEX_CREATE_MEMORY_USAGE;
 use crate::read::Batch;
 use crate::region::options::IndexOptions;
 use crate::sst::file::FileId;
 use crate::sst::index::intermediate::IntermediateManager;
+use crate::sst::index::inverted_index::creator::SstIndexCreator as InvertedIndexer;
+
+/// Output of the index creation.
+#[derive(Debug, Clone, Default)]
+pub struct IndexOutput {
+    /// Size of the file.
+    pub file_size: u64,
+    /// Inverted index output.
+    pub inverted_index: InvertedIndexOutput,
+}
 
-const INDEX_BLOB_TYPE: &str = "greptime-inverted-index-v1";
+/// Output of the inverted index creation.
+#[derive(Debug, Clone, Default)]
+pub struct InvertedIndexOutput {
+    /// Whether the index is available.
+    pub available: bool,
```

Comment:
If we can use `index_size` to check whether the index is available, do we still need the `available` flag?